doc/design/OrionMySQLSink.md

```diff
@@ -1 +1,172 @@
+3. The context responses/entities within the container are iterated, and a table called as the concatenation of `<fiware_servicePath>_<destination>` is created (if not yet existing) for each unit data.
+4. The context attributes within each context response/entity are iterated, and a new data row (or rows) is inserted in the current table. The format for this row depends on the configured persistence mode:
+    * `row`: A data row is added for each notified context attribute. This kind of row will always contain 8 fields:
+        * `recvTimeTs`: UTC timestamp expressed in miliseconds.
+        * `recvTime`: UTC timestamp in human-redable format ([ISO 6801](http://en.wikipedia.org/wiki/ISO_8601)).
+        * `entityId`: Notified entity identifier.
+        * `entityType`: Notified entity type.
+        * `attrName`: Notified attribute name.
+        * `attrType`: Notified attribute type.
+        * `attrValue`: In its simplest form, this value is just a string, but since Orion 0.11.0 it can be Json object or Json array.
+        * `attrMd`: It contains a string serialization of the metadata array for the attribute in Json (if the attribute hasn't metadata, an empty array `[]` is inserted).
+    * `column`: A single data row is added for all the notified context attributes. This kind of row will contain two fields per each entity's attribute (one for the value, called `<attrName>`, and other for the metadata, called `<attrName>_md`), plus an additional field about the reception time of the data (`recvTime`).
+
+####Important notes regarding the persistence mode
+Please observe not always the same number of attributes is notified; this depends on the subscription made to the NGSI-like sender. This is not a problem for the `row` persistence mode, since fixed 8-fields data rows are inserted for each notified attribute. Nevertheless, the `column` mode may be affected by several data rows of different lengths (in term of fields). Thus, the `column` mode is only recommended if your subscription is designed for always sending the same attributes, event if they were not updated since the last notification.
+
+In addition, when running in `column` mode, due to the number of notified attributes (and therefore the number of fields to be written within the Datastore) is unknown by Cygnus, the table can not be automatically created, and must be provisioned previously to the Cygnus execution. That's not the case of the `row` mode since the number of fields to be written is always constant, independently of the number of notified attributes.
+
+###Example
+Assuming the following Flume event is created from a notified NGSI context data (the code below is an <i>object representation</i>, not any real data format):
+
+    flume-event={
+        headers={
+	        content-type=application/json,
+	        fiware-service=vehicles,
+	        fiware-servicepath=4wheels,
```

Comment:
No. As I explained when asked for the same issue in the README: this is not the notified version of fiware-servicePath but the escaped one that is included within the internal Flume event header (in that case, the initial '/' is deleted).